cli/command/container/stop.go

```diff
@@ -39,16 +40,45 @@ func NewStopCommand(dockerCli *command.DockerCli) *cobra.Command {
 
 func runStop(dockerCli *command.DockerCli, opts *stopOptions) error {
 	ctx := context.Background()
+	timeout := time.Duration(opts.time) * time.Second
 
+	var waitAll sync.WaitGroup
 	var errs []string
+	const maxParallel int32 = 50
+
+	// make sure result is printed in order
+	output := map[string]chan error{}
 	for _, container := range opts.containers {
-		timeout := time.Duration(opts.time) * time.Second
-		if err := dockerCli.Client().ContainerStop(ctx, container, &timeout); err != nil {
-			errs = append(errs, err.Error())
-		} else {
-			fmt.Fprintf(dockerCli.Out(), "%s\n", container)
+		output[container] = make(chan error, 1)
+	}
+	waitAll.Add(1)
+	go func() {
+		for _, container := range opts.containers {
+			if err := <-output[container]; err != nil {
+				errs = append(errs, err.Error())
+			} else {
+				fmt.Fprintf(dockerCli.Out(), "%s\n", container)
+			}
 		}
+		waitAll.Done()
+	}()
```

Comment:
`Plus you might have to wait for them all to finish if the first container id is the last to finish, so you wouldn't really save any time.`

This is the worst situation. It's true that printing is fast, but stopping task may take long(at most 10 seconds), so I hope we can give user some output as quickly as possible, then user would know the stopping is really in progress but not stuck or dead.

So I would say it's really very beneficial.